scripts/update_and_rebuild_petsc.sh

```diff
@@ -81,7 +88,6 @@ if [ -z "$go_fast" ]; then
       --with-fc=mpif90 \
       --download-fblaslapack=1 \
       --download-metis=1 \
-      --download-ptscotch=1 \
```

Comment:
What is the intention here? I may misunderstand something. But I definitely would like to keep `ptscoth` that I use often. Unless there is a reason that we can not carry `ptscoth`

Reply:
When I try to build PETSc using this script with ptscotch included, it says ptscotch requires BISON, of all things, to be installed.  This package is also not included on the manual install instructions, and I haven't had any issue running BISON or any other herd code without it.  Is there a work-around that I'm unaware of?
![image](https://user-images.githubusercontent.com/21989132/63380837-370c6200-c34c-11e9-888c-21f49c272822.png)

Reply:
haha - This isn't BISON (the fuels performance code). This is bison (https://www.gnu.org/software/bison/). You can just get that through your normal package manager. @milljm - Do we need to add this to our instructions?

Reply:
Heh, that makes a _lot_ more sense.  I'll put that back as it was.  The packages I had to add from apt to make ptscotch happy are bison and flex.

Reply:
Most Linux installations seem to come with many packages I am about to list. But this should help @herter4171. I have to install the following when building our Singularity images (which start out as a minimalistic Ubuntu install):

```bash
 apt-get update
 apt-get -y upgrade
 apt-get -y install software-properties-common

 add-apt-repository "deb http://archive.ubuntu.com/ubuntu xenial main restricted universe multiverse"
 add-apt-repository "deb http://archive.ubuntu.com/ubuntu xenial-updates main restricted universe multiverse"
 apt-get update
 apt-get -y upgrade

apt-get -y install openssh-client libarchive-dev squashfs-tools \
  build-essential autoconf automake libtool debootstrap \
  python-dev software-properties-common curl libcurl4-gnutls-dev \
  patch gfortran flex bison tcl-dev tk-dev python-dev libblas-dev \
  liblapack-dev freeglut3-dev libxt-dev libncurses5-dev libedit-dev \
  zlib1g-dev libperl-dev lsb-release lsb-core libfftw3-bin libfftw3-dev \
  libfftw3-double3 libfftw3-long3 libfftw3-quad3 libfftw3-single3 vim \
  emacs libgsl-dev gawk man glpk-doc glpk-utils libglpk-dev csh tcsh \
  libtirpc-dev liblzma-dev guile-2.0-dev libxml++2.6-dev libcppunit-1.13 \
  libcppunit-dev screen less python-dev tcl-dev tk-dev swig libexpat1-dev \
  libssl-dev libxext-dev libxft-dev autoconf automake dnsutils valgrind \
  tcl tk gcc make libtool libhwloc-dev python python3-dev python3-numpy
```

I imagine some of those packages imply others on that list already. I haven't weeded it out. But that is a complete up-to-date list of the necessary 'things' I have had to add, or what others have asked me to add to aid in their development when using one of our Singularity images.

Reply:
And for Ubuntu-18:

```bash
apt-get update
apt-get -y upgrade
apt-get -y install libarchive-dev squashfs-tools build-essential \
  debootstrap python-dev software-properties-common

add-apt-repository "deb http://archive.ubuntu.com/ubuntu bionic main restricted universe multiverse"
add-apt-repository "deb http://archive.ubuntu.com/ubuntu bionic-updates main restricted universe multiverse"
apt-get update
apt-get -y upgrade

apt-get install -y locales openssh-client gfortran tcl m4 freeglut3 \
  doxygen libblas-dev liblapack-dev libx11-dev libnuma-dev zlib1g-dev \
  libhwloc-dev xvfb valgrind time curl libcurl4-gnutls-dev patch flex \
  bison tcl-dev tk-dev libblas-dev liblapack-dev freeglut3-dev libxt-dev \
  libncurses5-dev libedit-dev python-numpy bc libfftw3-dev libgsl-dev \
  gawk man vim emacs-nox lsb-release lsb-core valgrind libgsl-dev gawk \
  man glpk-doc glpk-utils libglpk-dev libtirpc-dev liblzma-dev \
  guile-2.0-dev libcppunit-1.14 libcppunit-dev screen tcsh csh python3-dev \
  python3-numpy
```

Reply:
Thanks, @milljm.  Everything is working as it should now.